pkg/bindings/containers/containers.go

```diff
@@ -333,3 +339,125 @@ func ContainerInit(ctx context.Context, nameOrID string) error {
 	}
 	return response.Process(nil)
 }
+
+// Attach attaches to a running container
+func Attach(ctx context.Context, nameOrId string, detachKeys *string, logs, stream *bool, stdin, stdout, stderr chan string) error {
+	conn, err := bindings.GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	params := url.Values{}
+	if detachKeys != nil {
+		params.Add("detachKeys", *detachKeys)
+	}
+	if logs != nil {
+		params.Add("logs", fmt.Sprintf("%t", *logs))
+	}
+	if stream != nil {
+		params.Add("stream", fmt.Sprintf("%t", *stream))
+	}
+	if stdin != nil {
+		params.Add("stdin", "true")
+	}
+	if stdout != nil {
+		params.Add("stdout", "true")
+	}
+	if stderr != nil {
+		params.Add("stderr", "true")
+	}
+
+	response, err := conn.DoRequest(nil, http.MethodPost, "/containers/%s/attach", params, nameOrId)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	ctype := response.Header.Get("Content-Type")
+	upgrade := response.Header.Get("Connection")
+
+	buffer := make([]byte, 1024)
+	if ctype == "application/vnd.docker.raw-stream" && upgrade == "Upgrade" {
```

Comment:
I think that the correct course of action here is likely to inspect the container and determine if Terminal is set, and then proceed based on that